src/ImageSharp/Formats/Jpeg/Components/Block8x8F.cs

```diff
@@ -488,18 +488,30 @@ public void LoadFromInt16Scalar(ref Block8x8 source)
     /// <param name="value">Value to compare to.</param>
     public bool EqualsToScalar(int value)
     {
-        // TODO: Can we provide a Vector128 implementation for this?
-        if (Avx2.IsSupported)
+        if (Vector256.IsHardwareAccelerated)
         {
-            const int equalityMask = unchecked((int)0b1111_1111_1111_1111_1111_1111_1111_1111);
-
             Vector256<int> targetVector = Vector256.Create(value);
             ref Vector256<float> blockStride = ref this.V256_0;
 
             for (nuint i = 0; i < RowCount; i++)
             {
-                Vector256<int> areEqual = Avx2.CompareEqual(Avx.ConvertToVector256Int32WithTruncation(Unsafe.Add(ref this.V256_0, i)), targetVector);
-                if (Avx2.MoveMask(areEqual.AsByte()) != equalityMask)
+                if (!Vector256.EqualsAll(Vector256.ConvertToInt32(Unsafe.Add(ref this.V256_0, i)), targetVector))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        if (Vector128.IsHardwareAccelerated)
+        {
+            Vector128<int> targetVector = Vector128.Create(value);
+            ref Vector4 blockStride = ref this.V0L;
```

Comment:
Is `blockStride` intentionally unused?

Reply:
Badly named (copy/paste) but yeah, I'm pointing to the `Vector4` field at offset 0. I don't have explicit Vector128<float> fields but am considering adding them to avoid some of the To/From `Vector128` code.